tools/mmp/aot.cs

```diff
@@ -0,0 +1,180 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Xamarin.Bundler {
+
+	public interface IFileEnumerator
+	{
+		IEnumerable<string> Files { get; }
+		string RootDir { get; }
+	}
+
+	public class FileSystemEnumerator : IFileEnumerator
+	{
+		DirectoryInfo Info;
+		public IEnumerable<string> Files => Info.GetFiles ().Select (x => x.FullName);
+		public string RootDir { get; private set; }
+
+		public FileSystemEnumerator (string path)
+		{
+			RootDir = path;
+			Info = new DirectoryInfo (path);
+		}
+	}
+
+	public delegate int RunCommandDelegate (string path, string args, string[] env = null, StringBuilder output = null, bool suppressPrintOnErrors = false);
+
+	public class AOTCompiler
+	{
+		enum AotType {
+			Default,
+			None,
+			All,
+			SDK,
+			Explicit
+		}
+
+		AotType aotType = AotType.Default;
+		public bool IsAOT => aotType != AotType.Default && aotType != AotType.None; 
+
+		// Set to Key -> True when we've seen a given include/exclude during compile to catch errors
+		Dictionary <string, bool> includedAssemblies = new Dictionary <string, bool> ();
+		Dictionary <string, bool> excludedAssemblies = new Dictionary <string, bool> ();
+
+		// Allows tests to stub out actual compilation and parallelism
+		public RunCommandDelegate RunCommand { get; set; } = Driver.RunCommand; 
+		public ParallelOptions ParallelOptions { get; set; } = new ParallelOptions () { MaxDegreeOfParallelism = Driver.Concurrency };
+
+		public string Quote (string f) => Driver.Quote (f);
+
+		public void Parse (string options)
+		{
+			// Syntax - all,sdk,none or "" if explicit then optional list of +/-'ed assemblies
+			// Sections seperated by ,
+			foreach (var option in options.Split (',')) {
+				switch (option) {
+				case "none":
+					aotType = AotType.None;
+					continue;
+				case "all":
+					aotType = AotType.All;
+					continue;
+				case "sdk":
+					aotType = AotType.SDK;
+					continue;
+				}
+
+				if (option.StartsWith ("+", StringComparison.InvariantCulture)) {
```

Comment:
Use `Ordinal`, not `Invariant` - or use a `char` comparison, like `options [0] == '+'`, with an earlier check for empty strings (or ask `Split` not to include empty strings).